infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/GitUserProfileProvisioner.java

```diff
@@ -82,9 +82,9 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
   }
 
   private Map<String, String> getMapFromJsonObject(String json) {
-    Type stringMapType = new TypeToken<Map<String, String>>() {}.getType();
+    Type typeToken = new TypeToken<Map<String, Object>>() {}.getType();
 
-    return new Gson().fromJson(json, stringMapType);
+    return new Gson().fromJson(json, typeToken);
```

Comment:
I'm not sure about how this is done; it looks like an implicit cast to `Map<String, String>`.

Reply:
Yes, it's implicit, when we get parsed and implicitly casted map, we will look for two specific properties, that we're 100% sure that have string value. We can add additional checking for a string type, but it will reduce code readability. Wdyt?

Reply:
It's up to you. It's not clear to me from reading the code that we can be 100% it's a string value, but I'm also not familiar with all the details here.

Is it possible for me to set my `git.user.name` to `false`?

Reply:
Theoretically - yes, but it's hard to do, you need to update preferences by hand and escape the whole json object to place it as string value in user preferences. But when https://github.com/eclipse/che-theia/pull/424 will be merged, these two properties will be configured through the UI and there will be an only option to setup them as string value.

Reply:
I'll add a comment, describes the idea of this hack

Reply:
Please make that any error in this class does not prevent workspace from starting.
To notify a user about an error you can provision a warning into k8s Environment, not sure if it's handled by Theia - if no - feel free to register an issue.